lib/publiccloud/utils.pm

```diff
@@ -210,4 +211,29 @@ sub define_secret_variable {
     script_run("set +a");
 }
 
+# Get credentials from the Public Cloud micro service, which requires user
+# and password. The resulting json will be stored in a file.
+sub get_credentials {
+    my $url = get_var('PUBLIC_CLOUD_CREDENTIALS_URL');
+    my $user = get_required_var('_SECRET_PUBLIC_CLOUD_CREDENTIALS_USER');
+    my $pwd = get_required_var('_SECRET_PUBLIC_CLOUD_CREDENTIALS_PASS');
+    my $credentials_json = '/root/credentials.json';
```

Comment:
Perhaps we should use `/root/gce.json` or `/root/credentials/gce.json` to make it possible to fetch multiple credentials in a single test run, if ever needed.

Not that we expect to do so, but right now this appears to be a very cheap change.

Reply:
yes, for instance in helm tests we are using multiple kubernetes PC

Reply:
Note that this is a common function that also other providers will use. That's why I picked a generic name `credentials.json`. This is the json that the service returns, which is applicable also for AWS and AZ.